feat: Lua rewrite for Neovim as alternative implementation

Producdevity opened the pull request:
## Summary

This PR introduces a complete rewrite of the WakaTime plugin in Lua, specifically designed for Neovim. The new implementation leverages Neovim's native Lua API, async capabilities, and modern plugin architecture, providing a more maintainable and performant experience for Neovim users.

## Motivation

- **Performance & Integration:** Lua is Neovim's native scripting language, offering better performance and deeper integration with Neovim's APIs compared to Vimscript.
- **Maintainability:** Lua code is easier to read, maintain, and extend, making future contributions and bug fixes more accessible.
- **Async Support:** The rewrite takes advantage of Neovim's asynchronous job control, resulting in a smoother user experience.
- **Type Annotations:** The new codebase includes Lua type annotations for improved developer experience and editor integration.
- **Modern Neovim Features:** The plugin now supports configuration via Lua tables, works seamlessly with plugin managers like Lazy.nvim and packer.nvim, and provides improved error handling and logging.

## Key Changes

- All core plugin logic is now implemented in Lua (`lua/wakatime/init.lua`).
- Async operations use Neovim's job control for non-blocking heartbeats and CLI installation.
- Configuration is now handled via Lua tables, supporting `opts` for Lazy.nvim and `setup()` for other managers.
- Type annotations are included for better autocompletion and documentation.
- Help documentation (`doc/wakatime.txt`) has been updated for the new Lua-based workflow.
- Added a `.stylua.toml` for consistent code formatting.

## Compatibility

- This rewrite is intended for Neovim users (0.7+ recommended).
- The original Vimscript implementation remains available for Vim and legacy Neovim users.

## Notes

- This is **not** a drop-in replacement for Vim users, but an alternative implementation for Neovim.
- Please review the updated documentation and help files for installation and configuration instructions.

---

Feedback and suggestions are welcome.

alanhamlett commented:
We need to support backwards compatibility for Vim users, so this should be either a branch or a separate repo unless there's another way to support both Vim (vimscript) and Neovim (lua) in one repo?

Producdevity commented:
> We need to support backwards compatibility for Vim users, so this should be either a branch or a separate repo unless there's another way to support both Vim (vimscript) and Neovim (lua) in one repo?

That sounds painfully annoying to manage in one repo. I am happy to maintain an unofficial lua version or make a PR if you decide to have a separate repo for this. 

I did this mainly as an introduction to lua and to mess around with the language

alanhamlett commented:
If you revert your commit that deletes `plugin/wakatime.vim` then I'll make some changes to load the lua version if running in Neovim and load the VimScript version when running in classic Vim.

Producdevity commented:
Sure, sounds good! What should happen to the `README.md`? I reverted `plugin/wakatime.vim` for now, I updated the `README.md` so it matches the refactor, it's on my master branch for now and left this one untouched.

alanhamlett commented:
Try these terminal commands:

```
git reset 7b53ba7882ba28b91b571fa4ffb2755378086837
git checkout README.md
git add .
git commit -am "Lua rewrite for Neovim"
git push --force-with-lease
```

Producdevity commented:
Are there any changes required for this to get merged?
